common/rpc/encryption/localStoreTlsProvider.go

```diff
@@ -238,32 +239,45 @@ func newServerTLSConfig(
 	certProvider CertProvider,
 	perHostCertProviderMap PerHostCertProviderMap,
 	config *config.GroupTLS,
+	logger log.Logger,
 ) (*tls.Config, error) {
 
 	clientAuthRequired := config.Server.RequireClientAuth
-	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired)
+	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired, "", logger)
 	if err != nil {
 		return nil, err
 	}
 
 	tlsConfig.GetConfigForClient = func(c *tls.ClientHelloInfo) (*tls.Config, error) {
+
+		remoteAddress := c.Conn.RemoteAddr().String()
+		logger.Info("attempted incoming TLS connection", tag.Address(remoteAddress), tag.HostID(c.ServerName))
+
 		if perHostCertProviderMap != nil {
 			perHostCertProvider, hostClientAuthRequired, err := perHostCertProviderMap.GetCertProvider(c.ServerName)
 			if err != nil {
+				logger.Warn("cannot find a per-host provider for attempted incoming TLS connection",
+					tag.HostID(c.ServerName), tag.Address(remoteAddress))
 				return nil, err
 			}
 
 			if perHostCertProvider != nil {
-				return getServerTLSConfigFromCertProvider(perHostCertProvider, hostClientAuthRequired)
+				return getServerTLSConfigFromCertProvider(perHostCertProvider, hostClientAuthRequired, remoteAddress, logger)
 			}
-			return getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired)
+			return getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired, remoteAddress, logger)
 		}
-		return getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired)
+		return getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired, remoteAddress, logger)
```

Comment:
do we want to log the same warning on lines 267 and/or 269 as well as that indicates we could not find a per-host provider too?